bitnami/common/templates/_secrets.tpl

```diff
@@ -122,6 +122,32 @@ The order in which this function returns a secret password:
 {{- printf "%s" $password -}}
 {{- end -}}
 
+{{/*
+Reuses the value from an existing secret, otherwise sets its value to a default value.
+
+Usage:
+{{ include "common.secrets.lookup" (dict "secret" "secret-name" "key" "keyName" "defaultValue" .Values.myValue "context" $) }}
+
+Params:
+  - secret - String - Required - Name of the 'Secret' resource where the password is stored.
+  - key - String - Required - Name of the key in the secret.
+  - defaultValue - String - Required - The path to the validating value in the values.yaml, e.g: "mysql.password". Will pick first parameter with a defined value.
+  - context - Context - Required - Parent context.
+
+*/}}
+{{- define "common.secrets.lookup" -}}
+
+{{- $value := "" -}}
+{{- $defaultValue := required "\n'common.secrets.lookup': Argument 'defaultValue' missing or empty" .defaultValue -}}
+{{- $secretData := (lookup "v1" "Secret" $.context.Release.Namespace .secret).data -}}
+{{- if and $secretData (hasKey $secretData .key) -}}
+  {{- $value = index $secretData .key | quote -}}
+{{- else -}}
+  {{- $value = $defaultValue | toString | b64enc | quote -}}
```

Comment:
do we normally quote values in the helpers? Not sure if this could be problematic for certain cases where you need to add the password like, for example a connection string

Reply:
I used "common.secrets.passwords.manage" as reference.

I'm sorry I don't understand the connection string example, as the output of this function will always be a base64 encoded.

Although, the suggestion made me think of a different use case.

If we add `| quote` inside the helper, it wouldn't be possible to run `| b64dec` if needed. For example:

`
include "common.secrets.lookup" (dict "secret" $secretName "key" "ca.crt" "defaultValue" $ca.Cert "context" $) | b64dec
`
Can't think of when would that be needed, but I would prefer not to limit the helper usage by forcing the `| quote`, so I'm removing it.